src/omnisharp/logger.ts

```diff
@@ -0,0 +1,59 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+'use strict';
+
+export class Logger {
+    private _writer: (message: string) => void;
+    private _prefix: string;
+
+    private _indent: string = '    ';
+    private _indentLevel: number = 0;
+    private _atLineStart: boolean = false;
+
+    constructor(writer: (message: string) => void, prefix?: string) {
+        this._writer = writer;
+        this._prefix = prefix;
+    }
+
+    private _appendCore(message: string): void {
+        if (this._atLineStart) {
+            if (this._indentLevel > 0) {
```

Comment:
Possibly. Seems like overkill just to repeat some characters. Left pad expects a length to be passed. I suppose I could use it to generate the indent string like so:

```JavaScript
leftPad('', this._indentLevel * 4, ' ');
```

Not sure it's worth it. :smile: